pygmt/exceptions.py

```diff
@@ -51,3 +54,64 @@ class GMTImageComparisonFailure(AssertionError):  # noqa: N818
     """
     Raised when a comparison between two images fails.
     """
+
+
+class GMTValueError(GMTError):
+    """
+    Raised when an invalid value is passed to a function/method.
+
+    Parameters
+    ----------
+    value
+        The invalid value.
+    description
+        The description of the value.
+    choices
+        The valid choices for the value.
+    reason
+        The detailed reason why the value is invalid.
+
+    Examples
+    --------
+    >>> raise GMTValueError("invalid")
+    Traceback (most recent call last):
+        ...
+    pygmt.exceptions.GMTValueError: Invalid value: 'invalid'.
+    >>> raise GMTValueError("invalid", description="constant name")
+    Traceback (most recent call last):
+    ...
+    pygmt.exceptions.GMTValueError: Invalid constant name: 'invalid'.
+    >>> raise GMTValueError("invalid", choices=["a", "b", 1, 2])
+    Traceback (most recent call last):
+        ...
+    pygmt.exceptions.GMTValueError: Invalid value: 'invalid'. Expected one of: 'a', 'b', 1, 2.
+    >>> raise GMTValueError("invalid", choices=["a", 0, True, False, None])
+    Traceback (most recent call last):
+        ...
+    pygmt.exceptions.GMTValueError: Invalid value: 'invalid'. Expected one of: 'a', 0, True, False, None.
+
+    >>> from pygmt.enums import GridType
+    >>> raise GMTValueError("invalid", choices=GridType)
+    Traceback (most recent call last):
+        ...
+    pygmt.exceptions.GMTValueError: Invalid value: 'invalid'. Expected one of: <GridType.CARTESIAN: 0>, <GridType.GEOGRAPHIC: 1>.
+    >>> raise GMTValueError("invalid", reason="Explain why it's invalid.")
+    Traceback (most recent call last):
+        ...
+    pygmt.exceptions.GMTValueError: Invalid value: 'invalid'. Explain why it's invalid.
+    """  # noqa: W505
+
+    def __init__(
+        self,
+        value: Any,
+        /,
+        description: str = "value",
+        choices: Iterable[Any] | None = None,
+        reason: str | None = None,
+    ):
+        msg = f"Invalid {description}: {value!r}."
+        if choices:
+            msg += f" Expected one of: {', '.join(repr(c) for c in choices)}."
+        if reason:
+            msg += f" {reason}"
```

Comment:
These lines would probably be a good use of [t-strings](https://realpython.com/python-t-strings/) (Template strings, [PEP 750](https://peps.python.org/pep-0750/)), but it is for Python 3.14+